File: AtCoder/ABC/208/cpp/B-Factorial_Yen_Coin.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

ll P;

int main() {
  cin >> P;
  vector<ll> parms{1};
  for (ll i = 1; i < 15; ++i) {
    parms.push_back(parms[i - 1] * i);
  }
  ll res = 0;
  for (ll i = 14; i > 0; --i) {
    res += min(P / parms[i], 100LL);
    P %= parms[i];
  }
  printf("%lld\n", res);
}

File: AtCoder/ABC/208/python/B-Factorial_Yen_Coin.py
parms = [1]
for i in range(1, 15):
    parms.append(parms[-1] * i)

P = int(input())
res = 0
for i in range(14, 0, -1):
    res += min(100, P // parms[i])
    P %= parms[i]
print(res)
